### ISSUE TITLE
deny_execmem logic is inverted

### ISSUE THREAD

**Issue by RoyalOughtness** [fedora-selinux/selinux-policy#2489]
the naming of `deny_execmem` implies that if it is set to `true`, then execmem will be denied. However, the opposite is the case, for example:

```
tunable_policy(`deny_execmem',`',`
	allow rpm_script_t self:process execmem;
')
```

It's for this reason that `deny_execmem` is set to `false`, even though the intent and effect is to deny execmem:

```
gen_tunable(deny_execmem,false)
```

This is highly confusing and should be renamed to `allow_execmem` and remain `false` by default. This would accurately capture the intent: that allowing execmem is disabled/`false`.

**Comment by yrro** [comment]
```
$ sesearch -A -b deny_execmem
allow NetworkManager_dispatcher_custom_t NetworkManager_dispatcher_custom_t:process execmem; [ deny_execmem ]:False
[...]
```

I've always read that as "allow the rule if `deny_execmem` is off"; is that wrong?

**Comment by zpytela** [comment]
@yrro The switch seems not to be well documented, so go into sources and you will see:

https://github.com/fedora-selinux/selinux-policy/blob/rawhide/policy/modules/system/unconfined.if#L58-L62

**Comment by zpytela** [comment]
@RoyalOughtness I think the rules are correct: if deny_execmem is true, no rule applies; if false, the execmem permission is allowed.

**Comment by yrro** [comment]
I agree. M4's syntax is really difficult to read, but:

```
tunable_policy(`deny_execmem',`',`
		# Allow making anonymous memory executable, e.g. 
		# for runtime-code generation or executable stack.
		allow $1 self:process execmem;
	')
```

`$1` will be `deny_execmem`, `$2` will be the empty string and `$3` will be the allow rules. This is expanded by:

```
define(`tunable_policy',`
        gen_require(`
                declare_required_symbols(`$1')
        ')
        if (`$1') {
                $2
        ifelse(`$3',`',`',`} else {
                $3
        ')}
')
```

... my head hurts puzzling this out but... `$3` will be the allow rules. This doesn't equal the empty string (2nd argument to `ifelse`) therefore `ifelse` will ignore the 3rd argument and expand to the 4th argument, which places the allow rules in an `else` statement.

Therefore the rules are _allowed_ if `deny_execmem` is `off`. I think the original reporter didn't realise that the allow rules are passed as the _third_ parameter to `tunable_policy` and not the second.

**Comment by zpytela** [comment]
Agreed, m4 is difficult to read, empty quotes are easy to miss, sesearch with a boolean is not quite clear, allow logic is more understandable than deny, etc. This is the current state though.
